Tensile/KernelWriter.py

```diff
@@ -314,22 +335,31 @@ def makeSchedule(self, kernel, tensorParametersA, tensorParametersB, localWriteE
     else:
       # create a plan:
       itemsLWToSched = list(self.localWriteACode.items()) + list(self.localWriteBCode.items())
-      if 1:
-        # This counts the number of modules which contain a ds_write
-        # Scheduler below keeps all writes in the same module in same iteration
-        # so this is better match to what it is trying to do
-        writesToSched = sum(1 for item in itemsLWToSched if item.countType(Code.LocalWriteInst))
-      else:
-        # count the number of writes, this doesn't match how they are
-        # scheduled so pushes writes up too far
-        writesToSched = self.localWriteACode.countType(Code.LocalWriteInst) + \
-                     self.localWriteBCode.countType(Code.LocalWriteInst)
+      # piggyback each global read instruction after each local write instruction to maximize G2L buffer utilization
+      if kernel["PrefetchGlobalRead"] == 2 or kernel["DepthULdsDivisor"]>1:
```

Comment:
recommend not to interleave longer latency memory instructions with shorter memory latency instructions. you really want to issue longer memory instructions b2b and issue all of them for better memory pipelining efficiency

Reply:
Will try that in the ensuing optimization work